internal/cli/auth/login.go

```diff
@@ -256,18 +272,18 @@ func newRegenerationPrompt() survey.Prompt {
 	}
 }
 
-func (opts *LoginOpts) LoginPreRun(ctx context.Context, c LoginConfig) func() error {
+func (opts *LoginOpts) LoginPreRun(ctx context.Context) func() error {
 	return func() error {
-		opts.config = c
-		if config.OpsManagerURL() != "" {
-			opts.OpsManagerURL = config.OpsManagerURL()
-		}
 		// ignore expired tokens since logging in
-		if err := opts.RefreshAccessToken(ctx); err != nil && !errors.Is(err, cli.ErrInvalidRefreshToken) {
-			return err
+		if err := opts.RefreshAccessToken(ctx); err != nil {
+			// clean up any expired or invalid tokens
+			opts.config.Set(config.AccessTokenField, "")
+
+			if !errors.Is(err, cli.ErrInvalidRefreshToken) {
+				return err
+			}
 		}
-		// clean up any expired tokens
-		opts.config.Set(config.AccessTokenField, "")
```

Comment:
moved this at line 280 otherwise login was continuing no matter if the user was logged in or not